### PR TITLE
bpo-33604: Bump removal notice from 3.6 to 3.8

### PR DESCRIPTION
**Pull request by Carreau**
Also change from PendingDeprecationWarning to DeprecationWarning.

https://bugs.python.org/issue33604

### CODE REVIEW

File: Lib/hmac.py [hunk 1]
```diff
@@ -50,7 +50,9 @@ def __init__(self, key, msg = None, digestmod = None):
 
         if digestmod is None:
             _warnings.warn("HMAC() without an explicit digestmod argument "
-                           "is deprecated.", PendingDeprecationWarning, 2)
+                           "is deprecated since Python 3.4, and will be removed "
```
Comment (gpshead) [inline Lib/hmac.py:53]: I'd be inclined to leave the warning text as is w/o mentioning version numbers there.  leave the versions to the docs.

**Review by gpshead — APPROVED** [review]
@ned-deily are we okay bringing this old PendingDeprecationWarning -> DeprecationWarning change into 3.7?

**Comment by ned-deily** [comment]
It's late in the game for 3.7 but, since we've said we were going to do in previous releases and there is an easy way to avoid the warning (just add an explicit digestmod), I'll go out on a limb and approve this for 3.7.0.

**Review by gpshead — APPROVED** [review]

**Comment by Carreau** [comment]
> It's late in the game for 3.7 but, since we've said we were going to do in previous releases and there is an easy way to avoid the warning (just add an explicit digestmod), I'll go out on a limb and approve this for 3.7.0.

Thanks I'v appreciate, extra kudos for being the release manager and doing all you are doing. 

> I'd be inclined to leave the warning text as is w/o mentioning version numbers there. leave the versions to the docs.

Ok, now that's it's merged I won't remove it, but personally when I see a deprecation, the version number since deprecation is about the only thing that is of interest, to know wether I need to have conditional code or not depending on Python version and how pressing it is. If I have to go pull the docs and look for that information I likely won't bother. It also make searching for deprecated features that need removal pretty easy...

Anyway thanks a gain and looking forward to to 3.7 !

**Pull request merged**